convert-mpt-hf-to-gguf.py

```diff
@@ -128,15 +128,21 @@ def parse_args() -> argparse.Namespace:
 # ref: https://github.com/cmp-nct/ggllm.cpp/blob/master/falcon_convert.py
 tokenizer = AutoTokenizer.from_pretrained(dir_model)
 
+added_vocab = tokenizer.get_added_vocab()
 reverse_vocab = {id: encoded_tok for encoded_tok, id in tokenizer.vocab.items()}
 
 for i in range(vocab_size):
-    tokens.append(reverse_vocab[i] if i in reverse_vocab else f"[PAD{i}]")
-    scores.append(0.0) # dummy
-    toktypes.append(gguf.TokenType.NORMAL)
+    if i in reverse_vocab:
+        tokens.append(reverse_vocab[i])
+        if reverse_vocab[i] not in added_vocab:
```

Comment:
It's more straightforward to check if `i` is in `added_vocab.values()`. See my version of the script (written before your BPE changes): https://github.com/nomic-ai/gpt4all/blob/37b007603ab818b596417edcbca0b3dd5e9bc85b/gpt4all-backend/scripts/convert_mpt_hf_to_gguf.py#L101-L103

Reply:
Tried to clarify the change.

Reply:
You're still using `elif reverse_vocab[i] in added_vocab:`, but my version uses `elif i in added_vocab:` and `tokenizer.get_added_vocab().values()`.

Reply:
Isn't this O(1) vs. O(n)?

Reply:
You're right, your version is more efficient if there are many added tokens. One could make a set of added token IDs, but I suppose it's simpler to do the dict lookup instead.